controllers/openstackcluster_controller.go

```diff
@@ -740,6 +740,33 @@ func reconcilePreExistingNetworkComponents(scope *scope.WithLogger, networkingSe
 		setClusterNetwork(openStackCluster, network)
 	}
 
+	// add pre-existing router to status if router spec is defined
+	if openStackCluster.Status.Router == nil {
+		openStackCluster.Status.Router = &infrav1.Router{}
+	}
```

Comment:
what happens if we don't do that?

Reply:
I dont remember why I added this. But when I read this now I think it doesnt make sense because it forces the user to provide a router reference otherwise they would get logs that the router couldnt be found over and over again.

Reply:
yeah this block doesn't make sense to me.